docs/source/data/fetching-data.mdx

```diff
@@ -1,6 +1,217 @@
+      return dataSources.dogsDB.getMostLikedDogs();
+    },
+    bigCats: async (_, __, { dataSources }) => {
+      return dataSources.catsApi.getCats({ size: 10 });
+    },
+  },
+};
+```
+
+## Open-source implementations
+
+[In Apollo Server 3](/apollo-server/v3/data/data-sources), you'd subclass the `DataSource` abstract class for each of your data sources. In Apollo Server 4, we recommend [making a custom class for each data source](#creating-data-source-classes), with each class best suited for that particular source of data.
+
+> ⚠️ **Note**: The community built each data source package below for use with Apollo Server 3. [As shown below](#using-datasource-subclasses), you can still use these packages in Apollo Server 4 with a bit of extra setup.
+
+The below data source implementations extend the generic [`DataSource` abstract class](https://github.com/apollographql/apollo-server/blob/main/packages/apollo-datasource/src/index.ts), from the deprecated `apollo-datasource` package. Subclasses of `DataSource` define the logic required to communicate with a particular store or API.
+
+The larger community maintains the following open-source implementations:
+
+| Class            | Source    | For Use With          |
+|------------------|-----------|-----------------------|
+| [`HTTPDataSource`](https://github.com/StarpTech/apollo-datasource-http)  | Community | HTTP/REST APIs |
+| [`SQLDataSource`](https://github.com/cvburgess/SQLDataSource)  | Community | SQL databases (via [Knex.js](http://knexjs.org/)) |
+| [`MongoDataSource`](https://github.com/GraphQLGuide/apollo-datasource-mongodb/) | Community | MongoDB |
+| [`CosmosDataSource`](https://github.com/andrejpk/apollo-datasource-cosmosdb) | Community | Azure Cosmos DB |
+| [`FirestoreDataSource`](https://github.com/swantzter/apollo-datasource-firestore) | Community | Cloud Firestore |
+
+> Apollo does not provide official support for community-maintained libraries. We cannot guarantee that community-maintained libraries adhere to best practices, or that they will continue to be maintained.
+
+### Using `DataSource` subclasses
+
+In Apollo Server 3, immediately after constructing each `DataSource` subclass, your server would invoke the `initialize({ cache, context })` method on each new `DataSource` behind the scenes.
+
+To replicate this in Apollo Sever 4, you can manually invoke the `initialize` method in the constructor function of each `DataSource` subclass, like so:
+
+<MultiCodeBlock>
+
+```ts
+import { ApolloServer } from '@apollo/server';
+import { startStandaloneServer } from '@apollo/server/standalone';
+import { KeyValueCache } from '@apollo/utils.keyvaluecache';
+import { Pool } from 'undici';
+import { HTTPDataSource } from 'apollo-datasource-http';
+
+class BooksAPI extends HTTPDataSource {
+  override baseURL = 'https://movies-api.example.com/';
+
+  constructor(options: { cache: KeyValueCache<string>; context: string }) { //highlight-line
```

Comment:
For some reason this is creating a new empty line below this one and highlighting it in the preview. Maybe has to do with the comment on line 175?

Reply:
Wait —in your deploy preview? The deploy preview looks fine for me?
![Screen Shot 2022-09-27 at 11 04 16 AM](https://user-images.githubusercontent.com/32436232/192602853-c8a3489f-cfbe-4d3a-aa5a-356e5d84357f.png)

Reply:
Fixed this!